flutter/lib/src/integrations/native_app_start_integration.dart

```diff
@@ -1,256 +1,29 @@
-// ignore_for_file: invalid_use_of_internal_member
-
-import 'dart:async';
-
 import 'package:meta/meta.dart';
 
 import '../../sentry_flutter.dart';
 import '../frame_callback_handler.dart';
-import '../native/sentry_native_binding.dart';
-import '../event_processor/native_app_start_event_processor.dart';
+import 'native_app_start_handler.dart';
 
-/// Integration which handles communication with native frameworks in order to
-/// enrich [SentryTransaction] objects with app start data for mobile vitals.
+/// Integration which calls [NativeAppStartHandler] after
+/// [SchedulerBinding.instance.addPostFrameCallback] is called.
 class NativeAppStartIntegration extends Integration<SentryFlutterOptions> {
-  NativeAppStartIntegration(this._native, this._frameCallbackHandler,
-      {Hub? hub})
-      : _hub = hub ?? HubAdapter();
+  NativeAppStartIntegration(
+      this._frameCallbackHandler, this._nativeAppStartHandler);
 
-  final SentryNativeBinding _native;
   final FrameCallbackHandler _frameCallbackHandler;
-  final Hub _hub;
+  final NativeAppStartHandler _nativeAppStartHandler;
 
   /// This timestamp marks the end of app startup. Either set automatically when
   /// [SentryFlutterOptions.autoAppStart] is true, or by calling
   /// [SentryFlutter.setAppStartEnd]
```

Comment:
I'm not sure if `setAppStartEnd` now makes sense anymore. what if the user decides to trigger setAppStartEnd after we have already sent the transaction then it has no effect

essentially it's only usable when the user decides the app start end is before our app start integration triggers it?

Reply:
as mentioned in the sync, we'll deprecate it after we've ensured that the user has a proper way to do this

so we have to be careful to keep the same behaviour